internal/pkg/runtime/engines/oci/prepare.go

```diff
@@ -171,21 +180,16 @@ func (e *EngineOperations) PrepareConfig(starterConfig *starter.Config) error {
 			e.EngineConfig.MasterPts = int(master.Fd())
 			e.EngineConfig.SlavePts = int(slave.Fd())
 		} else {
-			r, w, err := os.Pipe()
-			if err != nil {
+			flags := syscall.O_CLOEXEC
+			if err := syscall.Pipe2(e.EngineConfig.OutputStreams[0:], flags); err != nil {
```

Comment:
Can be simplified to `e.EngineConfig.OutputStreams[:]` to be more go-way

Reply:
Copied from https://golang.org/src/os/pipe_linux.go?s=319:360#L1 :)